src/contracts/core/DelegationManager.sol

```diff
@@ -528,132 +560,174 @@ contract DelegationManager is
+                        operator: podOwnerOperator,
+                        // the 'staker' here is the address receiving new shares
+                        staker: staker,
+                        strategy: withdrawal.strategies[i],
+                        scaledShares: increaseInDelegateableScaledShares
+                    });
                 }
-                unchecked {
-                    ++i;
+            } else {
+                strategyManager.addScaledShares(msg.sender, tokens[i], withdrawal.strategies[i], scaledShares[i]);
+                // Similar to `isDelegated` logic
+                if (currentOperator != address(0)) {
+                    _increaseOperatorScaledShares({
+                        operator: currentOperator,
+                        // the 'staker' here is the address receiving new shares
+                        staker: msg.sender,
+                        strategy: withdrawal.strategies[i],
+                        scaledShares: scaledShares[i]
+                    });
                 }
             }
+            unchecked {
+                ++i;
+            }
         }
-
-        emit WithdrawalCompleted(withdrawalRoot);
     }
 
-    // @notice Increases `operator`s delegated shares in `strategy` by `shares` and emits an `OperatorSharesIncreased` event
-    function _increaseOperatorShares(address operator, address staker, IStrategy strategy, uint256 shares) internal {
-        operatorShares[operator][strategy] += shares;
-        emit OperatorSharesIncreased(operator, staker, strategy, shares);
+    // @notice Increases `operator`s delegated scaled shares in `strategy` by `scaledShares` and emits an `OperatorSharesIncreased` event
+    function _increaseOperatorScaledShares(address operator, address staker, IStrategy strategy, uint256 scaledShares) internal {
+        operatorScaledShares[operator][strategy] += scaledShares;
+        // TODO: What to do about event wrt scaling?
+        emit OperatorSharesIncreased(operator, staker, strategy, scaledShares);
     }
 
-    // @notice Decreases `operator`s delegated shares in `strategy` by `shares` and emits an `OperatorSharesDecreased` event
-    function _decreaseOperatorShares(address operator, address staker, IStrategy strategy, uint256 shares) internal {
+    // @notice Decreases `operator`s delegated scaled shares in `strategy` by `scaledShares` and emits an `OperatorSharesDecreased` event
+    function _decreaseOperatorScaledShares(address operator, address staker, IStrategy strategy, uint256 scaledShares) internal {
         // This will revert on underflow, so no check needed
-        operatorShares[operator][strategy] -= shares;
-        emit OperatorSharesDecreased(operator, staker, strategy, shares);
+        operatorScaledShares[operator][strategy] -= scaledShares;
+        // TODO: What to do about event wrt scaling?
+        emit OperatorSharesDecreased(operator, staker, strategy, scaledShares);
```

Comment:
@ypatil12 @bowenli86 what do you think about pipeline/event concerns with slashing?

Reply:
Need to sit down and look through all events to make sure we are covered